### PR TITLE
New ipcheck

### PR DESCRIPTION
**Pull request by cledavid**

### CODE REVIEW

**Comment by cledavid** [comment]
resolves #11

File: ipchecking/ipChecking.go [hunk 1]
```diff
@@ -34,57 +32,53 @@ func StrToIP(iplist []string) ([]IP, error) {
...
 
+func isIPv4(ip string) bool {
+	for i := 0; i < len(ip); i++ {
+		switch ip[i] {
+		case '.':
+			return true
+		}
+	}
+	return false
+}
```
Comment (tomMoulard) [inline ipchecking/ipChecking.go:43]: Isn't there any better way to do this ? at least, don't use a switch for this, you can do this:
```go
if ip[i] == '.' {
    return true
}
```

File: ipchecking/ipChecking.go [hunk 2]
```diff
@@ -34,57 +32,53 @@ func StrToIP(iplist []string) ([]IP, error) {
...
 func (ip IP) CheckIPInSubnet(i string) bool {
-	checkIP, err := BuildIP(i)
-	if err != nil {
+	tempIP := net.ParseIP(i)
+	if tempIP == nil {
 		return false
 	}
-	return ip.IP>>(32-ip.Cidr) == checkIP.IP>>(32-ip.Cidr)
+	return ip.Net.Contains(tempIP)
 }
```
Comment (tomMoulard) [inline ipchecking/ipChecking.go:84]:
```go
func (ip IP) CheckIPInSubnet(i string) bool {
    return ip.Net.Contains(net.ParseIP(i))
}
```

File: ipchecking/ipChecking_test.go [hunk 3]
```diff
@@ -69,22 +69,85 @@ func TestIPGeneration(t *testing.T) {
...
+		{
+			name:     "[CIDR] Valid IPv6 CIDR ",
+			stringIP: "::1/53",
+			res:      true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := ipchecking.BuildIP(tt.stringIP)
```
Comment (tomMoulard) [inline ipchecking/ipChecking_test.go:127]: You don't test returned value :/

**Review by tomMoulard — CHANGES_REQUESTED** [review]

File: ipchecking/ipChecking.go (hunk 1 above)
Comment (cledavid) [reply ipchecking/ipChecking.go:43]: didn't find anything better than that so it'll stay like that and yeah i'll change for a if

Comment (tomMoulard) [reply ipchecking/ipChecking.go:43]: Can't you use `strings.Contains(ip, ".")` ?
https://golang.org/pkg/strings/#Contains

**Pull request merged**